test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1554,14 +1692,212 @@ def test_cgroup_assign_resources_mem_only_vnode(self):
+                           self.hostA.shortname, interval=3, offset=10)
+        result = self.server.status(NODE, 'resources_available.ncpus')
+        orig_ncpus = 0
+        for entry in result:
+            if entry['id'] == self.hostA.shortname:
+                orig_ncpus = int(entry['resources_available.ncpus'])
+                break
+        self.assertTrue(orig_ncpus > 0)
+        self.logger.info('Original value of ncpus: %d' % orig_ncpus)
+        if orig_ncpus < 2:
+            self.skipTest('Node must have at least two CPUs')
+        # Now exclude CPU zero
+        self.load_config(self.cfg5 % ('false', '0', 'false', 'false',
+                                      'false', self.swapctl))
+        self.server.expect(NODE, {'state': 'free'},
+                           self.hostA.shortname, interval=3, offset=10)
+        result = self.server.status(NODE, 'resources_available.ncpus')
+        new_ncpus = 0
+        for entry in result:
+            if entry['id'] == self.hostA.shortname:
+                new_ncpus = int(entry['resources_available.ncpus'])
+                break
+        self.assertTrue(new_ncpus > 0)
+        self.logger.info('New value with one CPU excluded: %d' % new_ncpus)
+        self.assertTrue((new_ncpus + 1) == orig_ncpus)
+        # Repeat the process with vnode_per_numa_node set to true
+        vnode = '%s[0]' % self.hostA.shortname
+        self.load_config(self.cfg5 % ('true', '', 'false', 'false',
+                                      'false', self.swapctl))
+        self.server.expect(NODE, {'state': 'free'},
+                           vnode, interval=3, offset=10)
+        result = self.server.status(NODE, 'resources_available.ncpus')
+        orig_ncpus = 0
+        for entry in result:
+            if entry['id'] == vnode:
+                orig_ncpus = int(entry['resources_available.ncpus'])
+                break
+        self.assertTrue(orig_ncpus > 0)
+        self.logger.info('Original value of vnode ncpus: %d' % orig_ncpus)
+        # Exclude CPU zero again
+        self.load_config(self.cfg5 % ('true', '0', 'false', 'false',
+                                      'false', self.swapctl))
+        self.server.expect(NODE, {'state': 'free'},
+                           vnode, interval=3, offset=10)
+        result = self.server.status(NODE, 'resources_available.ncpus')
+        new_ncpus = 0
+        for entry in result:
+            if entry['id'] == vnode:
+                new_ncpus = int(entry['resources_available.ncpus'])
+                break
+        self.assertTrue((new_ncpus + 1) == orig_ncpus)
+
+    def test_cgroup_cpuset_mem_fences(self):
+        """
+        Confirm that mem_fences affects setting of cpuset.mems
+        """
+        cpuset_base = self.get_cgroup_job_dir('cpuset', '123.foo', self.momA)
+        # Get the parent
+        cpuset_base = os.path.dirname(cpuset_base)
+        # Get the parent again
+        cpuset_base = os.path.dirname(cpuset_base)
```

Comment:
As above, change both these comments into one that says "Get the parent's parent"

Reply:
Done.